src/praisonai-agents/praisonaiagents/memory/adapters/factories.py

```
@@ -510,4 +551,166 @@ def close(self):
+        )
+        return [self._to_result(m) for m in response.memories]
+
+    # -- DeletableMemoryProtocol -------------------------------------------
+
+    def delete_memory(self, memory_id: str,
+                      memory_type: Optional[str] = None) -> bool:
+        """Delete a specific memory by id. Returns True on success."""
+        try:
+            self.client.forget(agent_id=self.agent_id, memory_id=memory_id)
+            return True
+        except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)